### ISSUE TITLE
Frost: Add support for Instagram handle

### ISSUE THREAD

**Issue by claesmk** [NCATComp410/comp410_fall_2022#62]
Finds all occurrences of an Instagram handle in a text string

**Comment by Amari-G** [comment]
Regex expression so far:

@[\w\d\.]{1,30}|\.

**Comment by claesmk** [comment]
What reference did you consult to determine which characters are valid at handles?

\d is a subset of \w so [\w\d] can be re-written as [\w] 

Be very careful with the period - the meaning is different if it is inside the [] vs outside of it - regex101 will explain what's happening.

**Comment by Amari-G** [comment]
I used regex101 while constructing the draft for the regex query. The regex I included in the comment works well for detecting valid usernames, but I still have some invalid  variations of usernames to try.

The comment was simply a way for me to "save my work", as I was constructing the regex query during class in a web browser.

With regards to your inquiry about the constraints that exist for Instagram usernames, the article at this [link](https://www.theverge.com/23176720/change-instagram-username-how-to)  states the following:

"Your username [...] can’t be longer than 30 characters and can only contain letters, numbers, periods, and underscores."
